roofit/roostats/src/RooStatsUtils.cxx

```diff
@@ -179,23 +173,24 @@ namespace RooStats {
          else return new RooProdPdf(TString::Format("%s_unconstrained", prod->GetName()).Data(),
             TString::Format("%s without constraints", prod->GetTitle()).Data(), newList);
 
-      } else if (id == typeid(RooExtendPdf)) {
+      } else if (dynamic_cast<RooExtendPdf*>(&pdf)) {
 
-         TIterator *iter = pdf.serverIterator();
+         auto iter = pdf.servers().begin();
          // extract underlying pdf which is extended; first server is the pdf; second server is the number of events variable
-         RooAbsPdf *uPdf = dynamic_cast<RooAbsPdf *>(iter->Next());
-         RooAbsReal *extended_term = dynamic_cast<RooAbsReal *>(iter->Next());
-         assert(uPdf != NULL); assert(extended_term != NULL); assert(iter->Next() == NULL);
-         delete iter;
+         auto uPdf = dynamic_cast<RooAbsPdf *>(*(iter++));
+         auto extended_term = dynamic_cast<RooAbsReal *>(*(iter++));
```

Comment:
I didn't check, but I suggest to make them `const` if possible. RooFit doesn't do `const`ness very well, but we are in the process of improving that.